CMakeLists.txt

```diff
@@ -1,551 +1,565 @@
-  # Silence appfat.cpp warnings
-  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-narrowing")
-endif()
-
-if(APPLE)
-  set_source_files_properties("./Packaging/macOS/AppIcon.icns" PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
-  set_source_files_properties("./Packaging/resources/CharisSILB.ttf" PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
-  set(MACOSX_BUNDLE_GUI_IDENTIFIER com.diasurgical.devilutionx)
-  set(MACOSX_BUNDLE_COPYRIGHT Unlicense)
-  set(MACOSX_BUNDLE_BUNDLE_NAME devilutionx)
-  set(MACOSX_BUNDLE_INFO_STRING ${PROJECT_VERSION})
-  set(MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION})
-  set(MACOSX_BUNDLE_SHORT_VERSION_STRING ${PROJECT_VERSION})
-  set(MACOSX_BUNDLE_LONG_VERSION_STRING "Version ${PROJECT_VERSION}")
-  set_target_properties(${BIN_TARGET} PROPERTIES MACOSX_BUNDLE_ICON_FILE "AppIcon")
-  set_target_properties(${BIN_TARGET} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/macOS/Info.plist")
-
-  install (TARGETS ${BIN_TARGET} DESTINATION ./)
-
-  if(DIST)
-      install(CODE "
-        include(BundleUtilities)
-        fixup_bundle(${CMAKE_BINARY_DIR}/${MACOSX_BUNDLE_BUNDLE_NAME}.app \"\" \"\")
-        "
-        COMPONENT Runtime)
-  endif()
-
-  set(MACOSX_BUNDLE_LONG_VERSION_STRING "Version ${PROJECT_VERSION}")
-  set(CPACK_PACKAGE_FILE_NAME "devilutionx")
-  set(CPACK_DMG_DISABLE_APPLICATIONS_SYMLINK "ON")
-  set(CPACK_GENERATOR "DragNDrop")
-
-  include(CPack)
-endif()
-
-if(SWITCH)
-  set(APP_TITLE "DevilutionX")
-  set(APP_AUTHOR "Devilution Team")
-  set(APP_ICON "${PROJECT_SOURCE_DIR}/Packaging/switch/icon.jpg")
-  set(APP_VERSION ${PROJECT_VERSION})
-  include(nx-utils)
-  build_switch_binaries(${BIN_TARGET})
-endif()
-
-if(RETROFW OR RG350)
-  set_target_properties(${BIN_TARGET} PROPERTIES OUTPUT_NAME "devilutionx.dge")
-endif()
+cmake_minimum_required(VERSION 3.8) # CMP0083 NEW
+
+include(CMake/out_of_tree.cmake)
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Debug")
+endif()
+
+option(ASAN "Enable address sanitizer" ON)
+option(UBSAN "Enable undefined behaviour sanitizer" ON)
+option(DEBUG "Enable debug mode in engine" ON)
+option(PIE "Generate position-independent code" OFF)
+option(LTO "Enable link-time optimization (if supported by the toolchain)" OFF)
+option(SPAWN "Build the shareware version" OFF)
+option(DIST "Dynamically link only glibc and SDL2" OFF)
+option(FASTER "Enable FASTER in engine" ON)
+option(BINARY_RELEASE "Enable options for binary release" OFF)
+option(NIGHTLY_BUILD "Enable options for nightly build" OFF)
+option(USE_SDL1 "Use SDL1.2 instead of SDL2" OFF)
+option(NONET "Disable network" OFF)
+
+include(CMake/git.cmake)
+get_git_tag(GIT_TAG)
+if(NOT CMAKE_BUILD_TYPE MATCHES "Release")
+  get_git_commit_hash(GIT_COMMIT_HASH)
+endif()
+
+project(DevilutionX
+  VERSION ${GIT_TAG}
+  LANGUAGES C CXX)
+
+if(BINARY_RELEASE)
+  set(CMAKE_BUILD_TYPE "Release")
+  set(ASAN OFF)
+  set(UBSAN OFF)
+  set(DEBUG OFF)  
+    if(AMIGA)
+      set(LTO OFF)
```

Comment:
This might not be needed as it checks if LTO is available below

Reply:
Cmake doesn't find it: Unknown CMake command "check_ipo_supported"